### PR TITLE
Add Player Sprint/Speed POC

### PR DESCRIPTION
**Pull request by tkwang0530**
- Implemented sprint functionality by modifying the Player class to handle different acceleration and max speed values when the shift key is held down.
- Added new attributes to the Player class: is_sprinting, normal_acceleration, sprint_acceleration, normal_max_speed, and sprint_max_speed.
- Added friction concept

### CODE REVIEW

File: examples/speed_poc.cc [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+        }
+        if (rect.x + SIZE > NUM_COL * SIZE) {
+            rect.x = NUM_COL * SIZE - SIZE;
+            vx = 0;
+        }
+    }
+};
+int main(int argc, char* argv[]) {
+    // Init music player
+    Mix_Init(0);
```
Comment (novayo) [inline examples/speed_poc.cc:155]: Please adjust to use sound manager

File: examples/speed_poc.cc [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+    float max_speed;
+
+    // is_sprinting: whether the player is sprinting
+    bool is_sprinting;
+
+    Player(int x, int y, int r, int g, int b) : Block(x, y, r, g, b), vx(0), ax(0), max_speed(300.0f), is_sprinting(false) {}
+
+    // Handle player input for movement and sprinting
+    void apply_input(SDL_Keycode key, bool is_pressed) {
+        float normal_acceleration = 200.0f;
```
Comment (novayo) [inline examples/speed_poc.cc:93]: Maybe consider to put the constants to the private variable?

Comment (tkwang0530) [reply examples/speed_poc.cc:93]: good idea

**Review by novayo — APPROVED** [review]
LGTM!

**Pull request merged**